crates/ironrdp-graphics/src/pointer.rs

```diff
@@ -156,12 +208,17 @@ impl DecodedPointer {
                 } else if and_bit == 1 && color == [0xff, 0xff, 0xff, 0xff] {
                     // Inverted pixel.
 
-                    // Colors with alpha channel set to 0x00 are always invisible no matter their color
-                    // component. We could take advantage of that, and use a special color to represent
-                    // inverted pixels. For example, we could use [0xFF, 0xFF, 0xFF, 0x00] for such
-                    // purpose.
-                    bitmap_data.extend_from_slice(&[0xFF, 0xFF, 0xFF, 0x00]);
-                } else {
+                    if target.inverted_pixels_as_checkboard() {
+                        // Checkboard pattern is used to represent inverted pixels.
+                        if (row_idx + col_idx) % 2 == 0 {
+                            bitmap_data.extend_from_slice(&[0xff, 0xff, 0xff, 0xff]);
+                        } else {
+                            bitmap_data.extend_from_slice(&[0x00, 0x00, 0x00, 0xff]);
+                        }
+                    } else {
+                        bitmap_data.extend_from_slice(&target.inverted_pixels_color());
+                    }
```

Comment:
We can't push this outside the loop, because this condition is only relevant for "invisible" pixels, which are enumerated amd found inside this loop. Maybe I missunderstood the suggestion?

Reply:
To clarify: I’m not suggesting to remove all conditions from the loop, only `if target.inverted_pixels_as_checkboard` one. It must be possible to push it out of the loop, because `target` is an immutable binding, and the result of `inverted_pixels_as_checkboard` will never change, even on the next iteration. Here is one possible avenue for that:
```diff
@@ -180,6 +180,19 @@ impl DecodedPointer {

         let mut bitmap_data = Vec::new();

+        let compute_inverted_pixel: fn(u16, u16) -> [u8; 4] = if target.inverted_pixels_as_checkboard() {
+            |row_idx, col_idx| {
+                // Checkered pattern is used to represent inverted pixels.
+                if (row_idx + col_idx) % 2 == 0 {
+                    [0xff, 0xff, 0xff, 0xff]
+                } else {
+                    [0x00, 0x00, 0x00, 0xff]
+                }
+            }
+        } else {
+            |_, _| [0xFF, 0xFF, 0xFF, 0x00]
+        };
+
         for row_idx in 0..data.height {
             // For non-monochrome cursors we read strides from bottom to top
             let (mut xor_stride_cursor, mut and_stride_cursor) = if flip_vertical {
@@ -207,17 +220,7 @@ impl DecodedPointer {
                     bitmap_data.extend_from_slice(&[0, 0, 0, 0]);
                 } else if and_bit == 1 && color == [0xff, 0xff, 0xff, 0xff] {
                     // Inverted pixel.
-
-                    if target.inverted_pixels_as_checkboard() {
-                        // Checkboard pattern is used to represent inverted pixels.
-                        if (row_idx + col_idx) % 2 == 0 {
-                            bitmap_data.extend_from_slice(&[0xff, 0xff, 0xff, 0xff]);
-                        } else {
-                            bitmap_data.extend_from_slice(&[0x00, 0x00, 0x00, 0xff]);
-                        }
-                    } else {
-                        bitmap_data.extend_from_slice(&target.inverted_pixels_color());
-                    }
+                    bitmap_data.extend_from_slice(&compute_inverted_pixel(row_idx, col_idx));
                 } else if target.premultiply_alpha() {
                     // Calculate premultiplied alpha via integer arithmetic
                     let with_premultiplied_alpha = [
```

Reply:
Ah, I see the idea now, thanks for the clarification